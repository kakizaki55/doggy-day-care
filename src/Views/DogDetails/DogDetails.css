.dog-detail-page {
  background-image: url('../../assests/cool-background.svg');
  height: 560px;
  background-color: rgb(152, 191, 225);
}
