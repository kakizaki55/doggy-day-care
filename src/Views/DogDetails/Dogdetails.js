import React from 'react';
import { useState, useEffect } from 'react';
import { fetchDogById } from '../../services/fetchDogs';
import DogDetailsCard from '../../components/DogDetailsCard/DogDetailsCard';
import './DogDetails.css';

export default function Dogdetails(props) {
  const [dog, setdog] = useState({});
  const [loading, setLoading] = useState(true);
  const id = props.match.params.id;

  useEffect(() => {
    const fetchData = async () => {
      const data = await fetchDogById(id);
      await setdog(data);
      setLoading(false);
    };
    fetchData();
  }, [id]);
  if (loading) {
    return <div>loading...</div>;
  } else {
    return (
      <div className="dog-detail-page">
        <DogDetailsCard {...dog} />
      </div>
    );
  }
}
