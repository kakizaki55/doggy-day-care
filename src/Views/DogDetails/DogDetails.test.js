import { render, screen } from '@testing-library/react';
import DogDetails from './Dogdetails';
import { MemoryRouter, Route } from 'react-router-dom';

it('span shot test for the home View container', async () => {
  const { container } = await render(
    <MemoryRouter initialEntries={['/dog/11']}>
      <Route exact path="/dog/:id" component={DogDetails}></Route>
    </MemoryRouter>
  );
  await screen.findByText('Meet Happy !');
  expect(container).toMatchSnapshot();
});
