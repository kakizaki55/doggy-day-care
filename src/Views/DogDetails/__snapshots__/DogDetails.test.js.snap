// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`span shot test for the home View container 1`] = `
<div>
  <div
    class="dog-details-card"
  >
    <h1>
       Meet 
      Happy
       ! 
    </h1>
    <img
      src="https://placedog.net/500?id=21"
    />
    <p>
      Happy
       is a(n) 
      11
       year old 
      French Bulldog
       
    </p>
    <p>
      A sweet but stinky guy. 
    </p>
    <a
      href="/"
    >
      back to all the Doggos
    </a>
  </div>
</div>
`;
